### DISCUSSION TITLE
Displaying relation values from 2 relationships deep on a select component

### DISCUSSION THREAD

**Discussion by scvn1** [filamentphp/filament#6768] [Help]
I have this on a resource's form:
```
                Forms\Components\Select::make('author_id')
                    ->relationship('author', 'id'),
```

I want the value displayed in the select list to actually be `->relationship('author', 'user.name')` from the `user()` relation on the `Author` model but this doesn't work.

How would I do something similar to this so I see a list of user names instead of IDs on the author model?

**Comment by danharrin** [comment]
Try
```
                Forms\Components\Select::make('author_id')
                    ->relationship('author', 'name', fn ($query) => $query->with('user'))
                    ->getOptionLabelFromRecordUsing(fn ($record) => $record->user->name),
```

**Reply by scvn1** [reply]
Unfortunately this produces an error:
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'name' in 'order clause'
select * from `authors` where `authors`.`deleted_at` is null order by `name` asc
```

**Reply by danharrin** [reply]
You aren't using `searchable()`, right?

**Reply by scvn1** [reply]
Nope...added it to my resource verbatim the way you posted it.
<img width="1018" alt="image" src="https://github.com/filamentphp/filament/assets/60627569/739f2c94-ddfe-4f47-bf0b-f1e52dee2dfd">

**Reply by danharrin** [reply]
Try chaining `->orderBy('authors.id')` onto the query

**Reply by scvn1** [reply]
That did it! Thanks!

**Reply by amhrn** [reply]
I was getting wrong IDs for the names since some "authors" might not have a "user" relationship (in my case). The following worked for me...
```
Select::make('author_id')
                    ->relationship('author', 'name', fn($query) => $query->whereHas('user'))
                    ->getOptionLabelFromRecordUsing(fn($record) => $record->user->name),
```
